pkg/config/config.go

```diff
@@ -178,17 +186,20 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if _, err := url.Parse(rc.APIURL); err != nil {
 			return fmt.Errorf("invalid api_url %q in receiver %q: %s", rc.APIURL, rc.Name, err)
 		}
-		if rc.User == "" {
-			if c.Defaults.User == "" {
-				return fmt.Errorf("missing user in receiver %q", rc.Name)
-			}
-			rc.User = c.Defaults.User
+
+		if rc.User != "" && rc.Password != "" && rc.PersonalAccessToken != "" {
+			return fmt.Errorf("bad auth config in receiver %q: user/password and PAT authentication are mutually exclusive", rc.Name)
 		}
-		if rc.Password == "" {
-			if c.Defaults.Password == "" {
-				return fmt.Errorf("missing password in receiver %q", rc.Name)
+
+		if (rc.User == "" || rc.Password == "") && rc.PersonalAccessToken == "" {
```

Comment:
That is weird - so we replace both password AND user if e.g user is set but password is not? Sounds wrong 🤔

Reply:
Hmmm, I was considering user & password as a block all the time, but it's true that being separate keys, they should be overridable separately. I will take a look to the logic again.

Reply:
Fixed in 107bc8f and added tests. User & Password can be overriden separately. The code has become a bit messy as there are a lot of passible scenarios to cover. I couldn't figure out something much cleaner 😓